pkg/controller/mizar/mizar-arktos-network-controller.go

```diff
@@ -206,3 +255,109 @@ func (c *MizarArktosNetworkController) processNetworkCreation(network *v1.Networ
+
+	// Get mapping from GVK for GVR (Group Version Resource) used by dynamic client resource
+	mapper := restmapper.NewDeferredDiscoveryRESTMapper(memory.NewMemCacheClient(discoveryClient))
+	klog.Infof("Mizar-Arktos-Network-controller - Name: %s - GVK group kind : (%v) - GVK version: (%v)", unstructuredObj.GetName(), gvk.GroupKind(), gvk.Version)
+	mapping, err := mapper.RESTMapping(gvk.GroupKind(), gvk.Version)
+
+	klog.V(5).Infof("Mizar-Arktos-Network-controller - Name: %s - GVK group kind : (%s) - GVK version: (%s)", unstructuredObj.GetName(), unstructuredObj.GetKind(), unstructuredObj.GetAPIVersion())
+	if err != nil {
+		klog.Errorf("Mizar-Arktos-Network-controller: get mapping between GVK and GVR in error (%v).", err)
+		return err
+	}
+
+	klog.V(5).Infof("Mizar-Arktos-Network-controller - Name: %s - get mapping scope name: (%s) - meta RESTScopeNameNamespace: (%s)", unstructuredObj.GetName(), mapping.Scope.Name(), meta.RESTScopeNameNamespace)
+
+	// Create dynamic client resource
+	var dynamicClientResource dynamic.ResourceInterface
+	if mapping.Scope.Name() == meta.RESTScopeNameNamespace {
+		// namespaced resources should specify the namespace under tenant
+		if unstructuredObj.GetNamespace() == "" {
+			unstructuredObj.SetNamespace("default")
+		}
+		namespace := unstructuredObj.GetNamespace()
+		klog.V(5).Infof("Mizar-Arktos-Network-controller - mapping resource: (%v) - set tenant: (%s) - namespace : (%s)", mapping.Resource, tenant, namespace)
+		dynamicClientResource = dynamicClient.Resource(mapping.Resource).NamespaceWithMultiTenancy(namespace, tenant)
+
+	} else {
+		// for cluster-wide resources
+		dynamicClientResource = dynamicClient.Resource(mapping.Resource)
+	}
+
+	// Create CRD resource - vpc or subnet
+	actualObject, err := dynamicClientResource.Create(unstructuredObj, metav1.CreateOptions{})
+
+	if err == nil {
+		klog.V(5).Infof("Mizar-Arktos-Network-controller - get actual object's name : (%s)", actualObject.GetName())
+		klog.V(5).Infof("Mizar-Arktos-Network-controller - get actual object's GVK : (%v)", actualObject.GroupVersionKind())
+		klog.V(5).Infof("Mizar-Arktos-Network-controller - get actual object's objectKind : (%v)", actualObject.GetObjectKind())
+	} else {
+		klog.Errorf("Mizar-Arktos-Network-controller - create actual object's name: (%s) in error (%v).", unstructuredObj.GetName(), err)
+	}
+
+	return err
+}
+
+func GetCRDVpcOrSubnetSpec(defaultTemplatePath, vpcOrSubnetName, tenant string) ([]byte, error) {
+	// For updating the data in default vpc/subnet template
+	var availableData = map[string]string{
+		"Tenant": tenant,
+	}
+
+	// Read template file
+	jsonTmpl, err := readTemplateFile(defaultTemplatePath)
+	if err != nil {
+		klog.Errorf("Mizar-Arktos-Network-controller - read default vpc/subnet template in error: (%v)", err)
+		return nil, err
+	}
+
+	// Create Template with template file
+	t := template.Must(template.New(vpcOrSubnetName).Parse(jsonTmpl))
```

Comment:
this op usually is expensive. consider refactor to only pay the cost once.

Reply:
Use "t, err := template.New(vpcOrSubnetName).Parse(tmpl)" instead.